EKrut-Server/src/ekrut/server/db/InventoryItemDAO.java

```diff
@@ -119,6 +119,41 @@ public Boolean updateItemThreshold(int itemId, String ekrutLocation, int thresho
 		}
 	}
 
+	/**
+	 * Updates the <b>threshold</b> of all inventory item(s) of a specific location.
+	 * 
+	 * @param area all thresholds in machines in that area will be updated. 
+	 * @param ekrutLocation the unique machine identifier. 
+	 * @return	True if update is successful, otherwise False.
+	 */
+	public Boolean updateItemsThresholdByArea(String area, int threshold) {
+		con.beginTransaction();
+		PreparedStatement ps = con.getPreparedStatement(
+				"UPDATE inventory_items "
+				+ "SET inventory_items.itemThreshold = ? "
+				+ "WHERE inventory_items.ekrutLocation IN "
+				+ "(SELECT ekrutLocation FROM machine_in_area WHERE area = ?);");
+		try {
+			ps.setInt(1, threshold);
+			ps.setString(3, area);
+			int count = con.executeUpdate(ps);
+			if (count != 1) {
```

Comment:
There can be more than one machine in an area. Therefore this operation may update more than one row.